Feature/homepage leaderboard

gscball opened the pull request.

gscball committed:
Email now Nickname, Leaderboard nearly done

gscball committed:
Leaderboard done

gscball committed:
UI tidy

gscball committed:
Final commit

bradleymiles17 commented on src/main/java/com/htisolutions/poolref/services/GreetingService.java:
```diff
@@ -0,0 +1,89 @@
+       leaderboard = removeUnusedUsers(leaderboard);
+       Collections.sort(leaderboard, new Comparator<LeaderboardEntry>() {
+           @Override
+           public int compare(LeaderboardEntry o1, LeaderboardEntry o2) {
```
you can do this with a bind operation in Java 8. It's much neater and only takes two lines or so

bradleymiles17 replied:
Like this:

Collections.sort(leaderboard, (a,b) -> a.getWins().compareTo(b.getWins()));

Commit:
- Refactor Leaderboard page to use ViewModel
- Remove Db access from view
- Move Db entities from model to entities directory
- BugFix: update email to nickname on RegisterController

Commit:
- Rename GreetingController to LeaderBoardController throughout site
- Remove static index view
- Apply mocking to Unit Test Service layer for all controller tests
